docroot/modules/custom/sitenow_advanced_webform/src/Plugin/WebformHandler/ProspectorRemotePostWebformHandler.php

```diff
@@ -0,0 +1,134 @@
+
+  /**
+   * The HTTP client to fetch the feed data with.
+   *
+   * @var \GuzzleHttp\ClientInterface
+   */
+  protected $httpClient;
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition) {
+    $instance = parent::create($container, $configuration, $plugin_id, $plugin_definition);
+    $instance->httpClient = $container->get('http_client');
+    return $instance;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitConfigurationForm(array &$form, FormStateInterface $form_state) {
+    parent::submitConfigurationForm($form, $form_state);
+    $this->applyFormStateToConfiguration($form_state);
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function prepareForm(
+    WebformSubmissionInterface $webform_submission,
+    $operation,
+    FormStateInterface $form_state,
+  ) {
+    // @todo This doesn't seem quite right because it might show to an end user.
+    //   Maybe it should be limited by the user's permissions? Or maybe there is
+    //   a better place to show this message where only a form admin would see
+    //   it?
+    // Load configuration.
+    $config = $this->configFactory->get('sitenow_advanced_webform.settings');
+    $site_uuid = $config->get('prospector.site_uuid');
+    // We need a site UUID to proceed.
+    if (!$site_uuid) {
+      // Generate a URL for the sitenow_advanced_webform settings form.
+      $url = Url::fromRoute('sitenow_advanced_webform.settings_form')->toString();
+      // Print a message directing the user to the settings form.
+      $this->messenger()->addError($this->t('Prospector Site UUID is missing. Please visit the settings form at @url to configure.', [
+        '@url' => $url,
+      ]));
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function postSave(WebformSubmissionInterface $webform_submission, $update = TRUE) {
+    // Load configuration.
+    $config = $this->configFactory->get('sitenow_advanced_webform.settings');
+    $site_uuid = $config->get('prospector.site_uuid');
+    // We need a site UUID to proceed.
+    if (!$site_uuid) {
+      // @todo Log this?
+      return;
+    }
+
+    // Load basic auth credentials from config.
+    $auth = $config->get('prospector.auth');
+    // We need auth credentials to proceed.
+    if (!isset($auth['user']) || !isset($auth['pass'])) {
+      // @todo I think this needs to be moved similar to the site_uuid check.
+      $this->messenger()->addError($this->t('Prospector authentication credentials are missing. Please contact the SiteNow team for assistance.'));
+      return;
+    }
+    $options = [
+      'auth' => array_values($auth),
+    ];
+
+    // Add data from first_name, last_name, email, and phone fields.
+    $elements = ['first_name', 'last_name', 'email', 'phone'];
+    $data = [];
+    foreach ($elements as $element) {
+      if ($value = $webform_submission->getElementData($element)) {
```

Comment:
does this return nicely if the field doesn't exist? with the field api we have to do a lot of that hasField type checking and i am less familiar with the webform elements.

Reply:
It does work if the element doesn't exist. I know this because I was testing against a form that used `email_address` and `phone_number`, which is another issue.